### PR TITLE
[Public Preview] Azure Confidential Ledger Extension

### PR DESCRIPTION
**Pull request by lynshi**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Confidential Ledger

File: src/confidentialledger/azext_confidentialledger/generated/commands.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        'azext_confidentialledger.vendored_sdks.confidentialledger.operations._ledger_operations#LedgerOperations.{}'
+    ),
+    client_factory=cf_ledger,
+)
+
+
+def load_command_table(self, _):
+
+    with self.command_group(
+        'confidentialledger', confidentialledger_ledger, client_factory=cf_ledger, is_experimental=True
```
Comment (lynshi) [inline src/confidentialledger/azext_confidentialledger/generated/commands.py:30]: What's the different between `is_experimental` and `is_preview` here?

Comment (lynshi) [reply src/confidentialledger/azext_confidentialledger/generated/commands.py:30]: Confirmed offline `is_experimental` is fine

File: src/confidentialledger/azext_confidentialledger/tests/latest/example_steps.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+# EXAMPLE: /Ledger/get/ConfidentialLedgerGet
+@try_manual
+def step_show(test, checks=None):
+    if checks is None:
+        checks = []
+    test.cmd('az confidentialledger show '
+             '--name "{myLedger}" '
+             '--resource-group "{rg}"',
+             checks=checks)
```
Comment (kairu-ms) [inline src/confidentialledger/azext_confidentialledger/tests/latest/example_steps.py:49]:
```suggestion
    return test.cmd('az confidentialledger show '
                              '--name "{myLedger}" '
                              '--resource-group "{rg}"',
                               checks=checks)
```

File: src/confidentialledger/azext_confidentialledger/tests/latest/test_confidentialledger_scenario.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+def call_scenario(test):
+    setup_scenario(test)
+    create_result = step_create(test, checks=[
+        test.check("location", "EastUS", case_sensitive=False),
+        test.check("name", "{myLedger}", case_sensitive=False),
+        test.check("properties.ledgerName", "{myLedger}", case_sensitive=False),
+        test.check("properties.ledgerType", "Public", case_sensitive=False),
+        test.check("tags.additionalProps1", "additional properties", case_sensitive=False),
+    ])
+    step_show(test, checks=[
```
Comment (kairu-ms) [inline src/confidentialledger/azext_confidentialledger/tests/latest/test_confidentialledger_scenario.py:53]:
```suggestion
    step_create(test, checks=[
        test.check("location", "EastUS", case_sensitive=False),
        test.check("name", "{myLedger}", case_sensitive=False),
        test.check("properties.ledgerName", "{myLedger}", case_sensitive=False),
        test.check("properties.ledgerType", "Public", case_sensitive=False),
        test.check("tags.additionalProps1", "additional properties", case_sensitive=False),
    ])
    create_result = step_show(test, checks=[
```

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**